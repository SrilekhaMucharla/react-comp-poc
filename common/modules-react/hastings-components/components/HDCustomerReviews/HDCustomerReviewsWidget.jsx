import PropTypes from 'prop-types';
import React, { useCallback, useEffect, useState } from 'react';
// import './HDCustomerReviews.scss';

const BREAKPOINT = 1200;
const RESIZE_EVENT = 'resize';
const REVIEWS_BADGE_ID = 'reviews-badge';

const HDCustomerReviewsWidget = ({
    onLoadReviewsBagde
}) => {
    const calcSize = useCallback(() => ((window.innerWidth < BREAKPOINT) ? 'small' : 'medium'));

    const [size, setSize] = useState(calcSize());

    useEffect(() => {
        const handleResize = () => {
            setSize(calcSize());
        };

        window.addEventListener(RESIZE_EVENT, handleResize);

        return () => window.removeEventListener(RESIZE_EVENT, handleResize);
    }, []);

    useEffect(() => {
        onLoadReviewsBagde(REVIEWS_BADGE_ID, size);
    }, [onLoadReviewsBagde, size]);

    return (
        <div className="customer-reviews-container">
            <div id="badge-horizontal" className="reviews-badge badge-horizontal" />
            <div id="badge-vertical" className="reviews-badge badge-vertical" />
        </div>
    );
};

HDCustomerReviewsWidget.propTypes = {
    onLoadReviewsBagde: PropTypes.func.isRequired,
};

export default HDCustomerReviewsWidget;
