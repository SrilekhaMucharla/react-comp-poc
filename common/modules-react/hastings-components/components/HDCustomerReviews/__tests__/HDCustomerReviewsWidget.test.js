import { shallow, mount } from 'enzyme';
import React from 'react';
import HDCustomerReviewsWidget from '../HDCustomerReviewsWidget';

describe('<HDCustomerReviewsWidget />', () => {
    test('render component', () => {
        const loadReviewsBagde = jest.fn();
        const wrapper = shallow((
            <HDCustomerReviewsWidget onLoadReviewsBagde={loadReviewsBagde} />
        ));
        expect(wrapper).toMatchSnapshot();
    });

    test('resize component', () => {
        const loadReviewsBagde = jest.fn();
        window.addEventListener = jest.fn();
        window.removeEventListener = jest.fn();

        const wrapper = mount((
            <HDCustomerReviewsWidget onLoadReviewsBagde={loadReviewsBagde} />
        ));
        expect(window.addEventListener).toBeCalled();

        wrapper.unmount();
        expect(window.removeEventListener).toBeCalled();
    });
});
