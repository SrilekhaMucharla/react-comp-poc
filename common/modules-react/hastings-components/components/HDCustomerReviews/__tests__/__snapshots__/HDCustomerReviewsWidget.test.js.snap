// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDCustomerReviewsWidget /> render component 1`] = `
<div
  className="customer-reviews-container"
>
  <div
    className="reviews-badge badge-horizontal"
    id="badge-horizontal"
  />
  <div
    className="reviews-badge badge-vertical"
    id="badge-vertical"
  />
</div>
`;
